metrics-operator/controllers/metrics/controller.go

```diff
@@ -101,23 +101,56 @@ func (r *KeptnMetricReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	reconcile := ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}
 
-	value, rawValue, err := r.getResults(ctx, metric, provider, metricProvider)
-	if err != nil {
-		r.Log.Error(err, "Failed to evaluate the query", "Response from provider was:", (string)(rawValue))
-		metric.Status.ErrMsg = err.Error()
-		metric.Status.Value = ""
-		metric.Status.RawValue = cupSize(rawValue)
-		metric.Status.LastUpdated = metav1.Time{Time: time.Now()}
-		reconcile = ctrl.Result{Requeue: false}
+	if metric.Spec.Range.StoredResults > 0 {
+		value, _, err := r.getResults(ctx, metric, provider, metricProvider)
+		if err != nil {
+			intervalResult := metricsapi.IntervalResult{
+				Value:       "",
```

Comment:
Value and Range do not need to specifically be set to their nil/zero values, so we can omit those two lines for easier readability